worker/src/scripts/upsertDefaultModelPrices.ts

```diff
@@ -0,0 +1,192 @@
+
+        if (
+          !force &&
+          existingModelUpdateDate &&
+          existingModelUpdateDate.getTime() ==
+            defaultModelPrice.updated_at.getTime()
+        ) {
+          logger.debug(
+            `Default model ${defaultModelPrice.model_name} (${defaultModelPrice.id}) already up to date. Skipping.`
+          );
+          continue;
+        }
+
+        // Upsert model and prices in a transaction
+        promises.push(
+          prisma
+            .$transaction(async (tx) => {
+              await tx.model.upsert({
+                where: {
+                  projectId: null,
+                  id: defaultModelPrice.id,
+                },
+                update: {
+                  modelName: defaultModelPrice.model_name,
+                  matchPattern: defaultModelPrice.match_pattern,
+                  updatedAt: defaultModelPrice.updated_at,
+                  tokenizerConfig:
+                    defaultModelPrice.tokenizer_config ?? undefined,
+                  tokenizerId: defaultModelPrice.tokenizer_id,
+                },
+                create: {
+                  projectId: null,
+                  id: defaultModelPrice.id,
+                  modelName: defaultModelPrice.model_name,
+                  matchPattern: defaultModelPrice.match_pattern,
+                  tokenizerConfig:
+                    defaultModelPrice.tokenizer_config ?? undefined,
+                  tokenizerId: defaultModelPrice.tokenizer_id,
+                  createdAt: defaultModelPrice.created_at,
+                  updatedAt: defaultModelPrice.updated_at,
+                },
+              });
+
+              for (const [usageType, price] of Object.entries(
+                defaultModelPrice.prices
+              )) {
+                await tx.price.upsert({
+                  where: {
+                    modelId_usageType: {
+                      modelId: defaultModelPrice.id,
+                      usageType,
+                    },
+                  },
+                  update: {
+                    price,
+                    updatedAt: defaultModelPrice.updated_at,
+                  },
+                  create: {
+                    modelId: defaultModelPrice.id,
+                    usageType,
+                    price,
+                    createdAt: defaultModelPrice.created_at,
+                    updatedAt: defaultModelPrice.updated_at,
+                  },
+                });
+              }
+
+              logger.info(
+                `Upserted default model ${defaultModelPrice.model_name} (${defaultModelPrice.id})`
+              );
+            })
+            .catch((error) => {
+              logger.error(
+                `Error upserting default model ${defaultModelPrice.model_name} (${defaultModelPrice.id}): ${error.message}`,
+                {
+                  error,
+                }
+              );
+            })
+        );
+      }
+
+      await Promise.all(promises);
```

Comment:
Promise.allSettled may be more applicable here. That way you do not put the fate of the entire batch on each single operation, but may perform all updates that are valid.

Reply:
Each promise/transaction has a catch clause, so `all` and `allSettled` are equivalent as they will all resolve